tables/output_tables.Rmd

```diff
@@ -13,132 +12,168 @@ library(openxlsx)
-  dplyr::select(Tumor_Sample_Barcode, tmb) %>%
-  dplyr::rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% 
-  dplyr::rename(Tmb_all = tmb) 
+tmb_all <- read_tsv(tmb_all_file) %>%
+  select(Tumor_Sample_Barcode, tmb) %>%
+  rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% 
+  rename(Tmb_all = tmb) 
 
 # read in tmb coding file, select and rename columns
-tmb_coding <- readr::read_tsv(file.path(data_dir, "pbta-snv-mutation-tmb-coding.tsv")) %>%
-  dplyr::select(Tumor_Sample_Barcode, tmb) %>%
-  dplyr::rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% 
-  dplyr::rename(Tmb_coding = tmb)
+tmb_coding <- read_tsv(tmb_coding_file) %>%
+  select(Tumor_Sample_Barcode, tmb) %>%
+  rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% 
+  rename(Tmb_coding = tmb)
 
 # combine files
-tmb_combined <- full_join(tmb_all, tmb_coding)  %>%
+tmb_combined <- full_join(tmb_all, tmb_coding) %>%
   mutate_all(as.character) %>% 
   replace(is.na(.), "NA")
 ```
 
 ### CNS mutational signatures
 ```{r s2 DNA results part 2}
-cns_mut_list <- readRDS("../analyses/mutational-signatures/results/fitted_exposures_signal-cns_sigfit.rds")
-cns_mean <- cns_mut_list[["mean"]] %>%
-  as.data.frame() %>%
-  tibble::rownames_to_column("Kids_First_Biospecimen_ID")  
+# use the deconstructSigs table, not the sigfit table, for consistency with main text
+# for convenience we can load the file deconstructsigs_exposures_merged.tsv which already has full sample information "merged" in _and_ uses the correct Signal RefSig names
+cns_mut_list <- read_tsv(mutsigs_file)
+
+cns_signatures <- cns_mut_list %>%
+  select(Kids_First_Biospecimen_ID, exposure, signature) %>%
+  # label signatures more clearly as RefSig or other
+  mutate(signature = ifelse(
+    signature == "Other", 
+    "Other signatures",
+    glue::glue("RefSig_{signature}")
+  )) %>%
+  spread(signature, exposure) %>%
+  # order columns so "Other" is at the end
+  select(Kids_First_Biospecimen_ID, contains("RefSig"), everything())
 
 ```
 
 ### Chromothripsis regions per sample
 ```{r s2 DNA results part 3}
-chromothripsis_region_df <- readr::read_tsv("../analyses/chromothripsis/results/chromothripsis_summary_per_sample.txt")  
+chromothripsis_region_df <- readr::read_tsv(chromo_file)  
 
 ```
 
 ### combine S2 table
 ```{r s2 DNA results combined}
-list_s2_table <- list(tmb = tmb_combined,
-                     cns_denovo_mut_sigs = cns_mean,
-                     chromothripsis_events = chromothripsis_region_df
+list_s2_table <- list(tumor_mutation_burden = tmb_combined,
+                      fitted_cns_mutational_signatures = cns_signatures,
+                      chromothripsis_events = chromothripsis_region_df
                      )
-openxlsx::write.xlsx(list_s2_table, 
-                     file.path(results_dir, "TableS2-DNA-results-table.xlsx"),
-                     overwrite=TRUE)
+write.xlsx(list_s2_table, 
+           table_s2_file,
+           overwrite=TRUE)
```

Comment:
So _sometimes_ we get errors with the cells `Number Stored as Text`... if I open this file locally in Excel, I see that the `Tmb_all` and `Tmb_coding` columns have these errors in the `tumor_mutation_burden` sheet. The `fitted_cns_mutational_signature` sheet has no such errors, despite the fact that the sheets both have IDs (expected to be text) and then floats. I wonder if `Tmb_all` and `Tmb_coding` are of class `character` for some reason and that's what's up?

Reply:
Oh: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1269/files#diff-96afe8a2465c3d1e8fe81bcb0bc6e065771299f64c624c1dcb0f6fa23ef3f7c6R99 Hm

Reply:
I've removed the `as.character()` business (which I think, but don't know for sure, was trying to work with NAs in Excel?) all around and added argument `keepNA=TRUE` to excel exports where we want NAs to show up instead of being changed to blank cells.